tests/RobotFramework/tests/cumulocity/smartrest_one/smartrest_one.robot

```diff
@@ -0,0 +1,78 @@
+*** Settings ***
+Resource            ../../../../resources/common.resource
+Library             Cumulocity
+Library             ThinEdgeIO
+
+Test Teardown       Custom Teardown
+
+Test Tags           theme:c8y    theme:operation
+
+
+*** Variables ***
+${SMART_REST_ONE_TEMPLATES}=
+...                             SEPARATOR=\n
+...                             10,339,GET,/identity/externalIds/c8y_Serial/%%,,application/vnd.com.nsn.cumulocity.externalId+json,%%,STRING,
+...                             10,311,GET,/alarm/alarms?source\=%%&status\=%%&pageSize\=100,,,%%,UNSIGNED STRING,
+...                             11,800,$.managedObject,,$.id
+...                             11,808,$.alarms,,$.id,$.type
+
+
+*** Test Cases ***
+Supports SmartREST 1.0 Templates
+    [Template]    Register and Use SmartREST 1.0. Templates
+    use_builtin_bridge=true
+    use_builtin_bridge=false
+
+
+*** Keywords ***
+Register and Use SmartREST 1.0. Templates
+    [Arguments]    ${use_builtin_bridge}
+    Custom Setup    use_builtin_bridge=${use_builtin_bridge}
+
+    ${TEMPLATE_XID}=    Get Random Name    prefix=TST_Template
+    Set Test Variable    $TEMPLATE_XID
+    Execute Command    tedge config set c8y.smartrest1.templates "${TEMPLATE_XID}"
+    Execute Command    tedge connect c8y    timeout=10
+    ${mo}=    Device Should Exist    ${DEVICE_SN}
+
+    # register templates
+    Execute Command
+    ...    curl --max-time 15 -sf -XPOST http://127.0.0.1:8001/c8y/s -H "Content-Type: plain/text" -H "X-Id: ${TEMPLATE_XID}" --data "${SMART_REST_ONE_TEMPLATES}"
+
+    # Use templates
+    # Get managed object id
+    Execute Command    cmd=tedge mqtt pub c8y/s/ul/${TEMPLATE_XID} '339,${DEVICE_SN}'
+    Should Have MQTT Messages    c8y/s/dl/${TEMPLATE_XID}    message_pattern=^800,\\d+,${mo["id"]}    timeout=10
+
+    Execute Command    cmd=tedge mqtt pub te/device/main///a/test '{"text":"test alarm","severity":"major"}' -r
+    Device Should Have Alarm/s    type=test    expected_text=test alarm
+
+    # Get alarms
+    Execute Command    cmd=tedge mqtt pub c8y/s/ul/${TEMPLATE_XID} '311,${mo["id"]},ACTIVE'
+    Should Have MQTT Messages    c8y/s/dl/${TEMPLATE_XID}    message_pattern=^808,\\d+,\\d+,test    timeout=10
+
+    # Operations
+    ${OPERATION}=    Get Configuration    tedge-configuration-plugin
```

Comment:
What's the point of this? Just making sure that other features are also working fine?

Reply:
Exactly. Testing if receiving operation from c8y works or not.

Reply:
More specifically, it tests the thin-edge.io components that rely on c8y API also work (e.g. uploading a file to Cumulocity).